File: sdekapi/warehouseservice/src/main/java/com/example/warehouseservice/warehouseservice/repository/WarehouseRepository.java
package com.example.warehouseservice.warehouseservice.repository;

import com.example.warehouseservice.warehouseservice.model.Warehouse;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.ExecutionException;

public interface WarehouseRepository {
    CompletableFuture<Boolean> addWarehouse(Warehouse warehouse) throws ExecutionException, InterruptedException;
    CompletableFuture<Boolean> updateWarehouse(Warehouse warehouse) throws ExecutionException, InterruptedException;
    CompletableFuture<Boolean> deleteWarehouse(Warehouse warehouse) throws ExecutionException, InterruptedException;
}

File: sdekflutterclient/lib/domain/repository/warehouse_repository.dart
import '../model/Warehouse.dart';

abstract class WarehouseRepository {
  Future<List<Warehouse>> getWarehouses();
  Future<Warehouse> getWarehouseById(int id);
  Future<void> addWarehouse(Warehouse warehouse);
  Future<void> updateWarehouse(Warehouse warehouse);
  Future<void> deleteWarehouse(Warehouse warehouse);
}